chia/wallet/did_wallet/did_wallet.py

```diff
@@ -794,6 +796,12 @@ async def create_message_spend(
             ],
             puzzle_announcements=puzzle_announcements,
             coin_announcements=coin_announcements,
+            coin_announcements_to_assert={a.name() for a in coin_announcements_to_assert}
```

Comment:
why do we require announcement object if we only need .name()? why not just bytes? same for puzzle ann

Reply:
I agree, I think this is something introduced a while ago, but it happens to be the standard for the API and I think we should prioritize consistency with the other transaction APIs to be fixed in a later refactor.